Remove opt-out of COM Reference Manager Tab

etbyrd opened the pull request:
This is a fix for: https://github.com/dotnet/project-system/issues/4355
I tested this locally and ensured that projects built with COM references but I am not sure if anything is blocking this on the SDK side.

Commit:
Remove opt-out of COM Reference Manager Tab

Pilchie commented:
Do we want to do this unilaterally, or only for projects where COM references could ever work? (.NET Framework or .NET Core >= 3.0)

etbyrd commented:
Most likely the latter - should this change go into a specific branch or should there be a check?

Pilchie commented:
There would have to be a check - the project system will support .NET Core < 3.0 and .NET Standard for a long time, so I would expect this code to need to support both projects where COM can work and ones where it can't.

Commit:
Update NETCoreApp Project Capabilities

etbyrd commented:
That makes sense. I updated to only remove for < 3.0

Pilchie reviewed (changes requested):
We should also remove it from `.NETStandard` libraries I think.  @jeffschwarz / @AaronRobinsonMSFT - Thoughts?

AaronRobinsonMSFT commented:
>We should also remove it from `.NETStandard` libraries I think.

@Pilchie This is annoying, but yes we should do that since using a .NETStandard assembly on a pre-3.0 CoreCLR will not yield a lot of success.

AaronRobinsonMSFT commented:
Thinking about this more, this is the right thing to do, because the binary will never be x-plat and so is very much not nor never will be .NETStandard. Good idea.

Commit:
Update for NETStandard projects

etbyrd commented:
Updated to remove in .NETStandard - how does this look?

Pilchie reviewed (approved):
LGTM - but need someone from the SDK team to take a look. @livarcocc @nguerrera ?

Pilchie commented:
Also tagging @davkean

nguerrera commented on src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:
```diff
@@ -600,13 +600,15 @@ Copyright (c) .NET Foundation. All rights reserved.
-  <!-- Reference Manager capabilities -->
+  <!-- NETCoreApp capabilities -->
   <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
     <ProjectCapability Remove="ReferenceManagerAssemblies" />
```
Existing: What does ReferenceManagerAssemblies do and why weren't we removing it for .NETStandard?

etbyrd replied:
These properties control the available tabs in the reference manager that is opened from the dependencies node context menu -> "Add Reference"
<img width="590" alt="refmang" src="https://user-images.githubusercontent.com/36282608/53598528-def16180-3b59-11e9-8802-0c93d392974b.PNG">

nguerrera replied:
Thanks. Looks like it's it just a bug that we're hiding this for .NET Core and not .NET Standard...

etbyrd replied:
I'm not sure - @Pilchie @AaronRobinsonMSFT  should we remove both assemblies and COM for .NET Standard?

nguerrera replied:
We can track this separately: https://github.com/dotnet/sdk/issues/2996

I just spotted it while reviewing your change, but it's existing so no need to come to a conclusion on this change.

AaronRobinsonMSFT replied:
>should we remove both assemblies and COM for .NET Standard?

@etbyrd I can only speak for COM and .NET Standard. Making the call for regular assemblies is above my pay grade.

@jeffschwMSFT can you provide some guidance on who could make the above decision?

davkean replied:
Yes we should remove this page for .NET Standard, however, making this change here will only take effect for the new SDK. We should move this into design-time targets on our side.

nguerrera commented on src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets:
```diff
@@ -600,13 +600,15 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectCapability Remove="ReferenceManagerAssemblies" />
+    <ProjectCapability Remove="ReferenceManagerCOM" Condition="$(_TargetFrameworkVersionWithoutV) &lt; '3.0'" />
+    <ProjectCapability Include="CrossPlatformExecutable" Condition="'$(_IsExecutable)' == 'true'" />
+  </ItemGroup>
```
@davkean Should these be moved to VS targets as in https://github.com/dotnet/project-system/pull/4400/files

We still don't seem to have clear guidance on when to put here and when to put there.

Commit:
Move back to just not removing the COM Reference capability here

etbyrd commented:
I updated to just not removing the COM Reference capability here, and added the rest of the behavior to the design time targets here: https://github.com/dotnet/project-system/pull/4654

etbyrd commented:
Closing to retry tests

etbyrd commented:
It looks like the failing tests are not related to this change - is this okay to merge? @nguerrera

nguerrera commented:
I thought we decided to move the handling to design time targets? https://github.com/dotnet/sdk/pull/2994#discussion_r261418282

nguerrera commented:
oh that was only for .net standard?

nguerrera commented:
@dsplaisted have you seen the mac failures here before?

nguerrera commented:
Sorry for the confusion in earlier, deleted messages. I'm caught back up now and reread above and project system PR. This looks good. I will follow up on #2996 separately with you.
